### PR TITLE
ENH: Add write_transform to store cache in GDAL location

### LINKED ISSUE #129: Store transform where GDAL will find it
https://gdal.org/drivers/raster/netcdf.html#georeference

```
 NetCDF driver will try to read the following metadata directly and set up georeferencing.
    spatial_ref (Well Known Text)
    GeoTransform (GeoTransform array)
```
It is probably worth doing for the case where `parse_coords=False` when reading a file and then it is written to netCDF. Also, I believe this is another part that goes along with #81

Currently, it is just stored in the `transform` attribute.

### LINKED ISSUE #139: Store transform
As described [here](https://corteva.github.io/rioxarray/stable/examples/manage_information_loss.html), information can be lost when doing xarray operations. Similar to the `write_crs` method, should we have a `write_transform` method that would ensure the transform gets propagated across operations? It also makes sense because some rioxarray operations, like `clip_box`, could update the transform.

### PR DESCRIPTION
**Pull request by snowman2**
- [x] Closes #129
-  [x] Closes #139
 - [x] Tests added
 - [x] Fully documented, including `docs/history.rst` for all changes and `docs/rioxarray.rst` for new API

- Noticed that the `sel()` operation returned an array that way sharing the coordinates of the parent for some reason ... so added `copy()` in `slice_xy()`.
- Noticed some test were using fixture instead of parametrize & weren't being run

### CODE REVIEW

File: rioxarray/rioxarray.py [hunk 1]
```diff
@@ -859,9 +917,12 @@ def transform_bounds(self, dst_crs, densify_pts=21, recalc=False):
         )
 
     def transform(self, recalc=False):
-        """Determine the affine of the `xarray.DataArray`"""
-        src_left, _, _, src_top = self.bounds(recalc=recalc)
-        src_resolution_x, src_resolution_y = self.resolution(recalc=recalc)
+        """affine.Afffine: The affine of the `xarray.DataArray`"""
```
Comment (jgruca) [inline rioxarray/rioxarray.py:920]: Is this docstring correct?

Comment (snowman2) [reply rioxarray/rioxarray.py]: Good catch - needed some updates.

**Comment by snowman2** [comment]
Appveyor failure seems to be unrelated. I am guessing to an update in scipy.

**Comment by snowman2** [comment]
Windows has scipy 1.5 & linux/osx has 1.4

**Pull request merged**